### PR TITLE
Various additions for mlibc on Linux project

### PR DESCRIPTION
**Pull request by 64**
This PR is part of the mlibc on Linux project.

### CODE REVIEW

**Comment by 64** [comment]
The `ELF_ST_*` changes are ABI breaks.

File: abis/linux/in.h [hunk 1]
```diff
@@ -44,6 +44,8 @@ struct ipv6_mreq {
 	unsigned        ipv6mr_interface;
 };
 
+extern const struct in6_addr in6addr_any;
```
Comment (ArsenArsen) [inline abis/linux/in.h:47]: where's the definition?

**Review by ArsenArsen — CHANGES_REQUESTED** [review]
lgtm past this and perhaps elaborating on reasons (and shortening the message) on 84caae12c8952b82e341d376a11cdce471e9026b

**Review by qookei — APPROVED** [review]
LGTM

**Review by ArsenArsen — APPROVED** [review]

**Review by Dennisbonke — APPROVED** [review]
LGTM

**Pull request merged**